addons/account/models/account_move.py

```diff
@@ -4569,11 +4600,11 @@ def write(self, vals):
                     raise UserError(_('You cannot modify the taxes related to a posted journal item, you should reset the journal entry to draft to do so.'))
 
             # Check the lock date.
-            if any(self.env['account.move']._field_will_change(line, vals, field_name) for field_name in PROTECTED_FIELDS_LOCK_DATE):
+            if line.parent_state == 'posted' and any(self.env['account.move']._field_will_change(line, vals, field_name) for field_name in PROTECTED_FIELDS_LOCK_DATE):
```

Comment:
From the look of it, the parent state is not in the vals. It is set to posted already when we update that line by giving it the move name, though in this case, it won't go in anyway as we only change the name.
After posting, the move.date should be correct since we can't post if it's not/it is automatically updated.